### PR TITLE
hpxMP release

### PR DESCRIPTION
**Pull request by tianyi93**
Update main CMake file for hpxMP configuration
Add hpxMP instructions to release procedure

### CODE REVIEW

File: docs/sphinx/contributing/release_procedure.rst [hunk 1]
```diff
@@ -51,6 +51,12 @@ are completed to avoid confusion.
      *   Update collaborators
      *   Update grant information
 
+#. Change the hpxMP release branch to be the most current release tag available
+   in the ``git_external`` hpxMP section of the main ``CMakeLists.txt``. Please
+   contact the maintainers at the `hpxMP repository
+   <https://github.com/STEllAR-GROUP/hpxMP>`_ to generate a new release to
+   synchronize with the HPX relelase.
```
Comment (msimberg) [inline docs/sphinx/contributing/release_procedure.rst]: Small typo: relelease.

Comment (tianyi93) [reply docs/sphinx/contributing/release_procedure.rst]: Thanks, fixed.
You might want to fix this typo for Apex release under this file as well.

**Comment by msimberg** [comment]
Yep, I'll take care of that.


On Thu, Nov 1, 2018, at 23:42, Tianyi Zhang wrote:
> *@tianyi93* commented on this pull request.


> 
> In docs/sphinx/contributing/release_procedure.rst[1]:


> > @@ -51,6 +51,12 @@ are completed to avoid confusion. *   Update
> > collaborators *   Update grant information  +#. Change the hpxMP
> > release branch to be the most current release tag available +   in
> > the ``git_external`` hpxMP section of the main ``CMakeLists.txt``.
> > Please +   contact the maintainers at the `hpxMP repository +
> > <https://github.com/STEllAR-GROUP/hpxMP>`_ to generate a new release
> > to +   synchronize with the HPX relelase.
>> Thanks, fixed.
>  You might want to fix this typo for Apex release under this file
>  as well.> — You are receiving this because your review was requested. Reply to
> this email directly, view it on GitHub[2], or mute the thread[3].

Links:

  1. https://github.com/STEllAR-GROUP/hpx/pull/3519#discussion_r230223268
  2. https://github.com/STEllAR-GROUP/hpx/pull/3519#discussion_r230223268

File: CMakeLists.txt [hunk 2]
```diff
@@ -642,11 +642,12 @@ hpx_option(HPX_WITH_ITTNOTIFY BOOL
 hpx_option(HPX_WITH_HPXMP BOOL
   "Enable hpxMP OpenMP emulation." OFF CATEGORY "OpenMP")
 if(HPX_WITH_HPXMP)
-  if(NOT MSVC AND ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"))
+  if(NOT MSVC AND (("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang") 
+	OR ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")))
     hpx_option(HPX_WITH_HPXMP_NO_UPDATE BOOL
       "Do not update code from remote hpxMP repository." OFF CATEGORY "OpenMP")
```
Comment (hkaiser) [inline CMakeLists.txt]: Can we change this message to something more appropriate, please?

File: CMakeLists.txt [hunk 3]
```diff
@@ -642,11 +642,12 @@ hpx_option(HPX_WITH_ITTNOTIFY BOOL
...
   else()
-    hpx_warn("hpxMP can be used only when compiling with clang")
+    hpx_warn("hpxMP can be used only when compiling with clang or gcc")
```
Comment (tianyi93) [inline CMakeLists.txt]: @hkaiser I guess this is the error message if we are not using clang or gcc, the one you commented on is a message when we define `HPX_WITH_HPXMP_NO_UPDATE `?

Comment (hkaiser) [reply CMakeLists.txt]: ok, I take it back. I misunderstood the code.

**Pull request merged**